src/types.rs

```diff
@@ -9,119 +9,96 @@ use std::{
 pub struct Guid(pub [u8; 16]);
 
 impl Guid {
-    /// Create new instance of Guid struct from a [0u8; 16] byte array
+    /// Create a guid from an array of sixteen u8s
     #[inline]
-    pub fn new(guid: [u8; 16]) -> Self {
-        Guid(guid)
+    pub const fn from_u8(value: [u8; 16]) -> Self {
+        Guid(value)
+    }
+
+    /// Create a guid from an array of four u32s
+    #[inline]
+    pub const fn from_u32(value: [u32; 4]) -> Self {
+        Guid(transmute_4u32_16u8(value))
+    }
+
+    /// Create a guid from a u128
+    #[inline]
+    pub const fn from_u128(value: u128) -> Self {
+        Guid(u128::to_le_bytes(value))
     }
 
     /// Returns true if the guid is zero.
     #[inline]
-    pub fn is_zero(&self) -> bool {
-        self.0.iter().all(|&x| x == 0)
+    pub const fn is_zero(&self) -> bool {
+        Guid::to_u128(self) == 0
+    }
+
+    /// Create an array of sixteen u8s from a guid
+    #[inline]
+    pub const fn to_u8(&self) -> [u8; 16] {
+        self.0
     }
 
-    /// Create a guid from a tuple of 4 ints
+    /// Create an array of four u32s from a guid
     #[inline]
-    pub const fn from_ints(value: (u32, u32, u32, u32)) -> Self {
-        let (a, b, c, d) = value;
-        Guid([
-            (a & 0xff) as u8,
-            ((a >> 8) & 0xff) as u8,
-            ((a >> 16) & 0xff) as u8,
-            ((a >> 24) & 0xff) as u8,
-            (b & 0xff) as u8,
-            ((b >> 8) & 0xff) as u8,
-            ((b >> 16) & 0xff) as u8,
-            ((b >> 24) & 0xff) as u8,
-            (c & 0xff) as u8,
-            ((c >> 8) & 0xff) as u8,
-            ((c >> 16) & 0xff) as u8,
-            ((c >> 24) & 0xff) as u8,
-            (d & 0xff) as u8,
-            ((d >> 8) & 0xff) as u8,
-            ((d >> 16) & 0xff) as u8,
-            ((d >> 24) & 0xff) as u8,
-        ])
+    pub const fn to_u32(&self) -> [u32; 4] {
+        transmute_16u8_4u32(self.0)
     }
+
+    /// Create a u128 from a guid
+    #[inline]
+    pub const fn to_u128(&self) -> u128 {
+        u128::from_le_bytes(self.0)
+    }
+}
+
+#[inline]
+const fn transmute_4u32_16u8(value: [u32; 4]) -> [u8; 16] {
+    let value_le = [
+        value[0].to_le(),
+        value[1].to_le(),
+        value[2].to_le(),
+        value[3].to_le(),
+    ];
```

Comment:
At debug levels this will cause some extra 32-bit copying of data, but in `--release` mode (`-C opt-level=3`), this gets completely optimized to 128bit instructions for both [u32;4] and [u8;16] types.